sdk/python/kfp/registry/registry_client.py

```diff
@@ -0,0 +1,512 @@
+            package_name_endpoint = f'{package_endpoint}/{{package_name}}'
+            tags_endpoint = f'{package_name_endpoint}/tags'
+            versions_endpoint = f'{package_name_endpoint}/versions'
+            config = {
+                'host':
+                    self._host,
+                'upload_url':
+                    self._host,
+                'download_version_url':
+                    f'{self._host}/{{package_name}}/{{version}}',
+                'download_tag_url':
+                    f'{self._host}/{{package_name}}/{{tag}}',
+                'get_package_url':
+                    f'{package_name_endpoint}',
+                'list_packages_url':
+                    package_endpoint,
+                'delete_package_url':
+                    f'{package_name_endpoint}',
+                'get_tag_url':
+                    f'{tags_endpoint}/{{tag}}',
+                'list_tags_url':
+                    f'{tags_endpoint}',
+                'delete_tag_url':
+                    f'{tags_endpoint}/{{tag}}',
+                'create_tag_url':
+                    f'{tags_endpoint}?tagId={{tag}}',
+                'update_tag_url':
+                    f'{tags_endpoint}/{{tag}}?updateMask=version',
+                'get_version_url':
+                    f'{versions_endpoint}/{{version}}',
+                'list_versions_url':
+                    f'{versions_endpoint}',
+                'delete_version_url':
+                    f'{versions_endpoint}/{{version}}',
+                'package_format':
+                    f'{repo_resource_format}/packages/{{package_name}}',
+                'tag_format':
+                    f'{repo_resource_format}/packages/{{package_name}}/tags/{{tag}}',
+                'version_format':
+                    f'{repo_resource_format}/packages/{{package_name}}/versions/{{version}}',
+            }
+        else:
+            logging.info(f'load_config not implemented for host: {self._host}')
+        return config
+
+    def _get_auth(self) -> requests.auth.AuthBase:
+        """Helper function to convert google credentials to AuthBase class if
+        needed.
+
+        Returns:
+            An instance of the AuthBase class
+        """
+        auth = self._auth
+        if isinstance(auth, credentials.Credentials):
+            auth = ApiAuth(auth.token)
+        return auth
+
+    def _refresh_creds(self) -> None:
+        """Helper function to refresh google credentials if needed."""
+        if self._is_ar_host() and isinstance(
+                self._auth, credentials.Credentials) and not self._auth.valid:
+            self._auth.refresh(google.auth.transport.requests.Request())
+
+    def upload_pipeline(self, file_name: str, tags: Optional[Union[str,
+                                                                   List[str]]],
+                        extra_headers: Optional[dict]) -> Tuple[str, str]:
+        """Upload the pipeline.
```

Comment:
nit: prefer descriptive-style to imperative-style throughout (ref: [Google Python styleguide](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods))

```suggestion
        """Uploads the pipeline.
```

Reply:
Done.